addons/stock/stock.py

```diff
@@ -4250,10 +4250,11 @@ def _complete_name(self, cr, uid, ids, name, args, context=None):
         """
         res = {}
         for m in self.browse(cr, uid, ids, context=context):
-            res[m.id] = m.name
+            res[m.id] = m.name or ''
             parent = m.parent_id
             while parent:
-                res[m.id] = parent.name + ' / ' + res[m.id]
+                parent_name = parent.name or ''
+                res[m.id] = "{} / {}".format(parent_name, res[m.id])
```

Comment:
Don’t use “.format” because it is not compatible with unicode string and show ugly errors.

You can use: “%s / %s” instead.

More info about:

https://www.odoo.com/documentation/9.0/reference/guidelines.html

“””
In general in Odoo, when manipulating strings, prefer % over .format() (when only one variable to replace in a string), and prefer %(varname) instead of position (when multiple variables have to be replaced).
“””